src/components/Icon/index.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './icon.scss';
+
+/**
+ * Base component for individual Icon
+ *
+ */
+export default class Icon extends React.Component {
+  /**
+    * @name Icon propTypes
+    * @type {propTypes}
+    *
+    * @param {Object} props - React PropTypes
+    *
+    * @property {string} icon -icon type to render based on the css
+    * @property {Function} clickHandler -handles icon click and keypress event
+    * @property {Boolean} active -determines if icon is active
+    *
+    * @return {Array} React propTypes
+    */
+  static propTypes = {
+    icon: PropTypes.string.isRequired,
+    clickHandler: PropTypes.func.isRequired,
+    active: PropTypes.bool
+  }
+
+  static defaultProps = {
+    active: false
+  }
+
+  /**
+    * handles icon click event and calls the attach clicked handler
+    * @method
+    *
+    * @return {void}
+    */
+  onClickHandler = () => {
+    const { clickHandler, ...otherProps } = this.props;
+
+    clickHandler(otherProps);
+  };
+
+  render() {
+    const { icon, active } = this.props;
+    return (
+      <button type="button" aria-label={icon} className={`icon icon-${icon} ${active && 'active'}`} onClick={this.onClickHandler} onKeyDown={this.onClickHandler} />
```

Comment:
We should use the appropriate modifiers both here and in the SCSS file. Remember we are using BEM

```suggestion
      <button type="button" aria-label={icon} className={`icon icon--${icon} ${active && 'icon--active'}`} onClick={this.onClickHandler} onKeyDown={this.onClickHandler} />
```